openedx/core/lib/xblock_builtin/xblock_discussion/xblock_discussion.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+"""
+Discussion XBlock
+"""
+import json
+import logging
+
+from xblockutils.resources import ResourceLoader
+from xblockutils.studio_editable import StudioEditableXBlockMixin
+
+from xblock.core import XBlock
+from xblock.fields import Scope, String, UNIQUE_ID
+from xblock.fragment import Fragment
+
+log = logging.getLogger(__name__)
+loader = ResourceLoader(__name__)  # pylint: disable=invalid-name
+
+
+def _(text):
+    """
+    A noop underscore function that marks strings for extraction.
+    """
+    return text
+
+
+@XBlock.needs('user')
+@XBlock.needs('i18n')
+class DiscussionXBlock(XBlock, StudioEditableXBlockMixin):
+    """
+    Provides a discussion forum that is inline with other content in the courseware.
+    """
+    discussion_id = String(scope=Scope.settings, default=UNIQUE_ID, force_export=True)
+    display_name = String(
+        display_name=_("Display Name"),
+        help=_("Display name for this module"),
+        default="Discussion",
+        scope=Scope.settings
+    )
+    discussion_category = String(
+        display_name=_("Category"),
+        default=_("Week 1"),
+        help=_(
+            "A category name for the discussion. "
+            "This name appears in the left pane of the discussion forum for the course."
+        ),
+        scope=Scope.settings
+    )
+    discussion_target = String(
+        display_name=_("Subcategory"),
+        default="Topic-Level Student-Visible Label",
+        help=_(
+            "A subcategory name for the discussion. "
+            "This name appears in the left pane of the discussion forum for the course."
+        ),
+        scope=Scope.settings
+    )
+    sort_key = String(scope=Scope.settings)
+
+    editable_fields = ["display_name", "discussion_category", "discussion_target"]
+
+    has_author_view = True  # Tells Studio to use author_view
+
+    @property
+    def course_key(self):
+        """
+        :return: int course id
+        """
+        return getattr(self.scope_ids.usage_id, 'course_key', None)
```

Comment:
@ormsbee CC @e-kolpakov I decided to leave the existing "cross-runtime" code, but [added a comment](https://github.com/edx/edx-platform/pull/12582/commits/934771f4da6f4978bf84b75ff657ca10bc497d46#diff-e3cb55b56190862761d9641bffe27ac9R66) explaining why it is the way it is.